src/beamlime/applications/_nexus_helpers.py

```diff
@@ -0,0 +1,303 @@
+            sub_dataset.clear()
+
+
+class NXDetectorContainer:
+    """NXDetector container.
+
+    This container is only for fake event data feeding.
+    """
+
+    def __init__(self, detector_dict: dict) -> None:
+        self.pixel_ids: list[int]
+        self.pixel_offsets: dict[str, list[float]]
+
+        self.detector_name: str = detector_dict['name']
+        self.detector_dict = detector_dict
+
+        self.init_pixel_ids()
+        self.init_pixel_offsets()
+
+    def init_pixel_ids(self) -> None:
+        """Initialize the pixel ids of the detector."""
+        pixel_id_filter = partial(_match_dataset_name, name='detector_number')
+        self._pixel_ids_idx = find_index(
+            self.detector_dict, 'children', pixel_id_filter, 'config', 'values'
+        )
+        self.pixel_ids = nested_dict_getitem(self.detector_dict, *self._pixel_ids_idx)
+
+    def init_pixel_offsets(self) -> None:
+        """Initialize the pixel offsets of the detector."""
+        pixel_offsets = [
+            child
+            for child in self.detector_dict['children']
+            if _match_dataset_name_suffix(child, 'pixel_offset')
+        ]
+        self.pixel_offsets = {
+            offset['config']['name']: offset['config']['values']
+            for offset in pixel_offsets
+        }
+        self._pixel_offsets_idx = {
+            offset['config']['name']: (
+                'children',
+                self.detector_dict['children'].index(offset),
+                'config',
+                'values',
+            )
+            for offset in pixel_offsets
+        }
+
+
+class NexusContainer:
+    nexus_template: dict
+
+    @classmethod
+    def from_template_file(cls, path: Path):
+        import json
+
+        with open(path) as f:
+            return cls(nexus_template=json.load(f))
+
+    def __init__(self, *, nexus_template: dict) -> None:
+        instrument_gr = self._get_instrument_group(nexus_template)
+        self.detectors = [
+            NXDetectorContainer(det_dict)
+            for det_dict in self._collect_detectors(instrument_gr)
+        ]
+
+    def _collect_detectors(self, instrument_gr: dict) -> list[dict]:
+        """Collect the detectors from the instrument group.
+
+        Only one or more NXdetectors are expected.
+        """
+        return [
+            child
+            for child in instrument_gr['children']
+            if match_nx_class(child, 'NXdetector')
+            # Expecting one or more NXdetector
+        ]
+
+    def _get_instrument_group(self, nexus_container: dict) -> dict:
+        """Get the NXinstrument group from the nexus container.
+
+        Only one NXinstrument group is expected.
+        """
+        return nested_dict_select(
+            nexus_container,
+            'children',
+            partial(match_nx_class, cls_name='NXentry'),
+            'children',
+            partial(match_nx_class, cls_name='NXinstrument'),
+        )
+
+    def prune_pixel_info(self) -> dict:
```

Comment:
Where is this used?

Reply:
It is a helper to prune pixel ids and pixel x/y/z offsets so that I can dump the json structure to another file and investigate it. The original files are too large so that the highlighter doesn't work, for example.

I guess we can just remove it though...

Reply:
Yes, please remove everything unrelated to the task.